sympy/simplify/fu.py

```diff
@@ -1075,7 +1076,10 @@ def handle_match(rv, num_args, den_args):
         rv = handle_match(rv, den_args, num_args)
         return rv
 
-    return bottom_up(rv, f)
+    if getattr(rv, "as_numer_denom()", False):
```

Comment:
Since such an object might be hiding in an expression -- `1 + Manifold?` -- it might be better to safe-guard the `f` with ```python
if not isinstance(rv, Expr):
    return rv
```
But I see that `bottom_up` has a `nonbasic` option. Perhaps it needs an `restict=None` keyword which would be a single class or tuple of classes to be used as a guard in `bottom_up`: if restrict and not isinstance(arg, restrict): return arg`.

Reply:
I bet there is already that type of filtering mechanism possible if you look at the files where `bottom_up` is defined.

Reply:
> Since such an object might be hiding in an expression -- `1 + Manifold?` -- it might be better to safe-guard the `f` with 
```python
>> if not isinstance(rv, Expr):
>>     return rv
```

Currently, this works well. But, should we also modify `bottom_up` according to your suggestion? 
Also, I think that using the keyword `restrict` and using the above code segment are more or less the same thing since `f` would not be applied if `isinstance(rv, Expr)` evaluates to be False. Please correct me if I am wrong.

Reply:
It's just that it could be used more generally. We have identified a need and that need is likely to rise in other places that use `bottom-up`. But I think this is ok for now.